src/shortcuts.js

```diff
@@ -84,100 +83,106 @@ const isGestureHint = state => state.alert && state.alert.alertType === 'gesture
 
 let handleGestureSegmentTimeout // eslint-disable-line fp/no-let
 
-/** Handles gesture hints when a valid segment is entered. */
-export const handleGestureSegment = (g, sequence, e) => {
-
-  const state = store.getState()
-  const { toolbarOverlay, scrollPrioritized } = state
-
-  if (toolbarOverlay || scrollPrioritized) return
-
-  // disable when modal is displayed or a drag is in progress
-  if (state.showModal || state.dragInProgress) return
-
-  const shortcut = shortcutGestureIndex[sequence]
-
-  // display gesture hint
-  clearTimeout(handleGestureSegmentTimeout)
-  handleGestureSegmentTimeout = setTimeout(
-    () => {
-      // only show "Invalid gesture" if hint is already being shown
-      alert(shortcut ? shortcut.name
-        : isGestureHint(state) ? '✗ Invalid gesture'
-        : null, { alertType: 'gestureHint', showCloseLink: false })
-    },
-    // if the hint is already being shown, do not wait to change the value
-    isGestureHint(state) ? 0 : GESTURE_SEGMENT_HINT_TIMEOUT
-  )
-}
-
-/** Executes a valid gesture and closes the gesture hint. */
-export const handleGestureEnd = (gesture, e) => {
-  const state = store.getState()
-  const { scrollPrioritized } = state
-
-  if (scrollPrioritized) return
-
-  // disable when modal is displayed or a drag is in progress
-  if (gesture && !state.showModal && !state.dragInProgress) {
-    const shortcut = shortcutGestureIndex[gesture]
-    if (shortcut) {
-      shortcutEmitter.trigger('shortcut', shortcut)
-      shortcut.exec(store.dispatch, store.getState, e, { type: 'gesture' })
-    }
-  }
-
-  // clear gesture hint
-  clearTimeout(handleGestureSegmentTimeout)
-  handleGestureSegmentTimeout = null // null the timer to track when it is running for handleGestureSegment
-
-  // needs to be delayed until the next tick otherwise there is a re-render which inadvertantly calls the automatic render focus in the Thought component.
-  setTimeout(() => {
-    if (isGestureHint(store.getState())) {
-      alert(null)
+/**
+ * Keyboard handlers factory function.
+ */
+export const keyboardHandlers = store => ({
+
+  /** Handles gesture hints when a valid segment is entered. */
+  handleGestureSegment: (g, sequence, e) => {
+
```

Comment:
Since `handleGestureSegment` and `handleGestureEnd` are not keyboard inputs, a better name would be `inputHandlers`.

Reply:
Sure!